modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/resources/BaseExtensionResourcesGeneratorAnnotationProcessor.java

```diff
@@ -88,9 +91,17 @@ public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment
         });
       });
 
+
       return false;
+    } catch (MuleRuntimeException e) {
+      if (e.getCause() instanceof IllegalModelDefinitionException) {
```

Comment:
why don't you use ExceptionUtils.findCause() or similar to have a more in depth search of causes?